### ISSUE TITLE
custom data trainning

### ISSUE THREAD

**Issue by zhangyunming** [LikeLy-Journey/SegmenTron#4]
hi ,it is a good job though only 2 stars , please continue to maintain the job
and give the custom dataload and config. haha ,thansks

**Comment by LikeLy-Journey** [comment]
> hi ,it is a good job though only 2 stars , please continue to maintain the job
> and give the custom dataload and config. haha ,thansks

I will try to be better.
About custom data, you can convert you data format into cityscape format. One picture correspond to a ground truth picture. And then use cityscape dataloader.

**Comment by zhangyunming** [comment]
hi, as you said ,i put my own data to cityscapes dir , but in the dataloder/cityscapes.py , my class is 2,labels are 0 and 1, how to change below:
thanks!


raise RuntimeError("Found 0 images in subfolders of:" + root + "\n")
        self.valid_classes = [7, 8, 11, 12, 13, 17, 19, 20, 21, 22,
                              23, 24, 25, 26, 27, 28, 31, 32, 33]
        self._key = np.array([-1, -1, -1, -1, -1, -1,
                              -1, -1, 0, 1, -1, -1,
                              2, 3, 4, -1, -1, -1,
                              5, -1, 6, 7, 8, 9,
                              10, 11, 12, 13, 14, 15,
                              -1, -1, 16, 17, 18])

**Comment by LikeLy-Journey** [comment]
sorry, it is a little complicated.
you should modify img_folder and  mask_folder [here](https://github.com/LikeLy-Journey/SegmenTron/blob/master/segmentron/data/dataloader/cityscapes.py#L121-L122) 。and then change [here](https://github.com/LikeLy-Journey/SegmenTron/blob/master/segmentron/data/dataloader/cityscapes.py#L110) to your folder to make sure you can find images.
finally, change self.valid_classes=[0,1], self._key = [0,1],

**Comment by zhangyunming** [comment]
thanks for reply
i changed what you said and can find the images data. but i think self._key = [0,1] is not right , you can see in the cityscapes the self._key nums is not 19.

**Comment by LikeLy-Journey** [comment]
> thanks for reply
> i changed what you said and can find the images data. but i think self._key = [0,1] is not right , you can see in the cityscapes the self._key nums is not 19.

valid key number is 19, not valid key set as -1. your key is all valid

**Comment by lichen14** [comment]
@LikeLy-Journey @zhangyunming  
Hi, can your explain more details on how to train(&eval) on custom dataset? thanks.

**Comment by andrew-begain** [comment]
> thanks for reply
> i changed what you said and can find the images data. but i think self._key = [0,1] is not right , you can see in the cityscapes the self._key nums is not 19.

@zhangyunming @lichen14 您好，麻烦问一下您在自己的数据集上训练模型成功了吗？我想根据自己的数据集训练deeplabv3+，dataloader使用的是pascal voc的，我的数据集不是21类，颜色也不是跟voc是一样的，麻烦问一下，我应该修改什么来适应我这种情况？期待大神回复，谢谢

**Comment by S534435877** [comment]
> > thanks for reply
> > i changed what you said and can find the images data. but i think self._key = [0,1] is not right , you can see in the cityscapes the self._key nums is not 19.
> 
> @zhangyunming @lichen14 您好，麻烦问一下您在自己的数据集上训练模型成功了吗？我想根据自己的数据集训练deeplabv3+，dataloader使用的是pascal voc的，我的数据集不是21类，颜色也不是跟voc是一样的，麻烦问一下，我应该修改什么来适应我这种情况？期待大神回复，谢谢

我将自己的数据集制作成cityscape格式，是可以训练的，https://blog.csdn.net/s534435877/article/details/106937561
self.valid_classes=[0,1], self._key = [-1,0,1]
未标注的为0，标注了一类为1.

**Comment by dreamlychina** [comment]
> > > thanks for reply
> > > i changed what you said and can find the images data. but i think self._key = [0,1] is not right , you can see in the cityscapes the self._key nums is not 19.
> > 
> > 
> > @zhangyunming @lichen14 您好，麻烦问一下您在自己的数据集上训练模型成功了吗？我想根据自己的数据集训练deeplabv3+，dataloader使用的是pascal voc的，我的数据集不是21类，颜色也不是跟voc是一样的，麻烦问一下，我应该修改什么来适应我这种情况？期待大神回复，谢谢
> 
> 我将自己的数据集制作成cityscape格式，是可以训练的，https://blog.csdn.net/s534435877/article/details/106937561
> self.valid_classes=[0,1], self._key = [-1,0,1]
> 未标注的为0，标注了一类为1.

我看了你的博客，脚本的最后要进行crop，cityscapes数据集必须得resize到固定的尺寸？？？

**Comment by dreamlychina** [comment]
> self.valid_classes=[0,1], self._key = [-1,0,1]

如果我的类别数4，self.valid_classes=[0,1], self._key = [-1,0,1]怎么填写？？self.valid_classes=[0,1，2，3，4，], self._key = [-1,0,1，2，3，4]对吗？大佬
